### ISSUE TITLE
question about your paper

### ISSUE THREAD

**Issue by xuyifan-0731** [WindyLee0822/Process_Q_Model#2]
Hello! The article "Process Reward Model with Q-Value Rankings" has been very inspiring to me, but I have some questions regarding the details of the paper, and I hope to receive clarifications from the authors.

A key aspect of the ranking loss calculation in the paper is the availability of known step-level labels. However, in the Math-Shepherd PRM dataset you used, these labels are not manually annotated (such as "let’s verify step by step," which can be considered almost entirely accurate). Instead, they are estimated through methods like MCTS. Do you think such labels could introduce additional errors?

In scenarios beyond mathematics, obtaining PRM labels might be quite challenging. For instance, in many agent scenarios, well-annotated step-level datasets are not available, and only trajectory-level rewards can be obtained. We might use some RL methods to estimate Q or Advance values to infer incorrect steps. However, since Q values are already used in your loss, I’m curious whether this approach could introduce additional errors, in your opinion.

Thank you again for this high-quality paper, and I look forward to your response.

**Comment by WindyLee0822** [comment]
Thank you so much for your interest in our paper! I am happy to clarify your questions!

> Do automatically annotated labels introduce additional errors?

Exactly. Prior research (e.g. Math-shepherd) treats all steps following the first incorrect step as wrong.  Hence, we adapt our theoretical loss (Eq. 9) to a practical version (Eq. 10), and also empirically compare these two objective in Section 4.3. We also include another objective variant in Appendix C, which emphasize the Q-value gap between correct steps and the first incorrect step rather than all wrong steps. The results in Table 3 and Table 6 reveals the existence of the noise in this regardless negative annotations. 

Indeed, the annotation of correct steps and the location of the first error can also be noisy. How to mitigate the annotation noise and collect higher-quality corpora for PRMs is a great direction for the future studies to explore. We believe the performance of PQM can be further enhanced by a higher-quality training data. 

> We might use some RL methods to estimate Q or Advance values to infer incorrect steps. However, since Q values are already used in your loss, I’m curious whether this approach could introduce additional errors, in your opinion.

Do you mean estimating Q or Advance values by implicit Q function as in [1]? I think our Q-value ranking relationships (Theorem 3.1) should also established in that setting. It worth a try, but might require some additional adaptions.

[1] Rafailov R, Hejna J, Park R, et al. From $ r $ to $ Q^* $: Your Language Model is Secretly a Q-Function[J]. arXiv preprint arXiv:2404.12358, 2024.

Feel free to reopen the issue if you have further questions!

**Comment by jkkjjj** [comment]
great work
